File: app/src/main/java/com/app/walletcards/model/ApplyCardResponse.kt
package com.app.walletcards.model

import kotlinx.serialization.Serializable

@Serializable
data class ApplyCardResponse(
    @Serializable(with = CodeSerializer::class)
    val code: String? = null,
    val status: String = "",
    val message: String = "",
    val depositaddress: String? = null,
    val subuserfee: Double? = null
)

File: mastercard/lib/data/models/apply_card_response.dart
class ApplyCardResponse {
  final String? code;
  final String status;
  final String message;
  final String? depositaddress;
  final double? subuserfee;

  ApplyCardResponse({
    this.code,
    this.status = "",
    this.message = "",
    this.depositaddress,
    this.subuserfee,
  });

  factory ApplyCardResponse.fromJson(Map<String, dynamic> json) {
    return ApplyCardResponse(
      code: json['code']?.toString(),
      status: json['status'] ?? "",
      message: json['message'] ?? "",
      depositaddress: json['depositaddress'],
      subuserfee: (json['subuserfee'] as num?)?.toDouble(),
    );
  }
}
